tests/integration/integration-test.sh

```diff
@@ -1,238 +1,305 @@
+uninstall() {
+  echo -n 'Uninstalling Connaisseur...'
+  make uninstall >/dev/null || {
+    echo -e "${FAILED}"
+    exit 1
+  }
+  echo -e "${SUCCESS}"
+}
+
+uninstall_helm() {
+  echo -n 'Uninstalling Connaisseur...'
+  helm uninstall connaisseur -n connaisseur &&
+    kubectl delete ns connaisseur >/dev/null || {
+    echo -e "${FAILED}"
+    exit 1
+  }
+  echo -e "${SUCCESS}"
+}
+
+update_values() {
+  for update in "$@"; do
+    yq e -i "${update}" helm/values.yaml
+  done
+}
+
+debug_vaules() {
+  echo "::group::values.yaml"
+  cat helm/values.yaml
+  echo "::endgroup::"
+}
+
+### RUN REGULAR INTEGRATION TEST ####################################
+regular_int_test() {
+  multi_test "regular"
+
+  ### EDGE CASE TAG IN RELEASES AND TARGETS ####################################
+  echo -n "[edge1] Testing edge case of tag defined in both targets and release json file..."
+  DEPLOYED_SHA=$(kubectl get pod pod-rs -o yaml | yq e '.spec.containers[0].image' - | sed 's/.*sha256://')
+  if [[ "${DEPLOYED_SHA}" != 'c5327b291d702719a26c6cf8cc93f72e7902df46547106a9930feda2c002a4a7' ]]; then
+    echo -e "${FAILED}"
+  else
+    echo -e "${SUCCESS}"
+  fi
+  N=$(($N + 1))
+
+  ### ALERTING TEST ####################################
+  echo -n "Checking whether alert endpoints have been called successfully..."
+  ENDPOINT_HITS="$(curl -s ${ALERTING_ENDPOINT_IP}:56243 --header 'Content-Type: application/json')"
+  NUMBER_OF_DEPLOYMENTS=$((${DEPLOYMENT_VALIDITY["VALID"]} + ${DEPLOYMENT_VALIDITY["INVALID"]}))
+  EXPECTED_ENDPOINT_HITS=$(jq -n \
+    --argjson REQUESTS_TO_SLACK_ENDPOINT ${NUMBER_OF_DEPLOYMENTS} \
+    --argjson REQUESTS_TO_OPSGENIE_ENDPOINT ${DEPLOYMENT_VALIDITY["VALID"]} \
+    --argjson REQUESTS_TO_KEYBASE_ENDPOINT ${DEPLOYMENT_VALIDITY["INVALID"]} \
+    '{
+  "successful_requests_to_slack_endpoint":$REQUESTS_TO_SLACK_ENDPOINT,
+  "successful_requests_to_opsgenie_endpoint": $REQUESTS_TO_OPSGENIE_ENDPOINT,
+  "successful_requests_to_keybase_endpoint": $REQUESTS_TO_KEYBASE_ENDPOINT
+  }')
+  diff <(echo "$ENDPOINT_HITS" | jq -S .) <(echo "$EXPECTED_ENDPOINT_HITS" | jq -S .) >diff.log 2>&1 || true
+  if [[ -s diff.log ]]; then
+    echo -e "${FAILED}"
+    echo "::group::Alerting endpoint diff:"
+    cat diff.log
+    echo "::endgroup::"
+    EXIT="1"
+  else
+    echo -e "${SUCCESS}"
+  fi
+}
+
+### COSIGN TEST ####################################
+cosign_int_test() {
+  multi_test "cosign"
+}
+
+### NAMESPACE VALIDATION TEST ####################################
+namespace_val_int_test() {
+  echo -n "Creating namespaces..."
+  kubectl create namespace ignoredns >/dev/null
+  kubectl label ns ignoredns securesystemsengineering.connaisseur/webhook=ignore >/dev/null
+  kubectl create namespace validatedns >/dev/null
+  kubectl label ns validatedns securesystemsengineering.connaisseur/webhook=validate >/dev/null
+  echo -e "${SUCCESS}"
+
+  multi_test "ignore-namespace-val"
+  update_values '.namespacedValidation.mode="validate"'
+  upgrade
```

Comment:
Why does namespace test need an upgrade?

Reply:
because the `namespacedValidation.mode` is switched from `ignore` to `validate`, which needs to be changed in the webhookconfiguration. that can only happen through an `helm upgrade`.

Reply:
As I always say, I'm not firm with the helm part. If that's obvious, feel free to ignore, otherwise add a comment